src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1252,6 +1252,15 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
     if (need_add_to_database)
         database = DatabaseCatalog::instance().tryGetDatabase(database_name);
 
+    if (database && database->getEngineName() == "Replicated" && create.select)
+    {
+        bool allow_create_select_for_replicated = create.isView() || create.is_create_empty;
+        if (!allow_create_select_for_replicated)
+            throw Exception(
+                ErrorCodes::SUPPORT_IS_DISABLED,
+                "CREATE AS SELECT is not supported with Replicated databases. Use separate CREATE and INSERT queries");
```

Comment:
Actually, CREATE SELECT makes sense for non-replicated tables in Replicated databases. Should we also check `table->supportsReplication()`? (but it's not necessary, the only reason to add this check is to reduce the number of tests that we have to disable)